server/src/main/java/org/cloudfoundry/identity/uaa/provider/oauth/ExternalOAuthAuthenticationManager.java

```diff
@@ -424,7 +424,7 @@ private String getMappedClaim(String externalName, String internalName, Map<Stri
             return (String) claimObject;
         }
         if (claimObject instanceof Collection) {
-            Set<String> entry = ((Collection<?>) claimObject).stream().map(String.class::cast).collect(Collectors.toSet());
+            Set<String> entry = ((Collection<?>) claimObject).stream().filter(String.class::isInstance).map(String.class::cast).collect(Collectors.toSet());
```

Comment:
Will collection items always going to be a string? Should we consider handle it in more generic way:

Set<String> entry = ((Collection<?>) claimObject).stream()
        .map(claim -> String.valueOf(claim))
        .collect(Collectors.toSet());

Reply:
I think it is not wanted to have such a generic approach, but simple check if the entries inside of the token are strings. With your approach we allow all but then we are incompatible with now. I mean if claims are type of Boolean or int we would allow them now. But that is not wanted